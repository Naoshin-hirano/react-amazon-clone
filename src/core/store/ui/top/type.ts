export type UITop = {
    productInCart: Busket;
    currentTotalAmount: number;
    currentProductNum: number;
};

export type Busket = {
    id: string;
    title: string;
    image: string;
    price: number;
    rating: number;
}[];

export type CartProduct = {
    id: string;
    title: string;
    image: string;
    price: number;
    rating: number;
};
