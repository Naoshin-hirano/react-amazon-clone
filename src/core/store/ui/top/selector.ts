import { createSelector } from "@reduxjs/toolkit";
import { StateType } from "core/store";
import { UITop } from "./type";

export const UITopSelector = createSelector(
    (state: ReturnType<StateType>) => state["ui/top"],
    (state: UITop) => state
);
