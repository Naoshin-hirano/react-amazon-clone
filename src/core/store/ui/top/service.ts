import { PayloadAction } from "@reduxjs/toolkit";
import { CartProduct, UITop } from "./type";

export const initialState = {
    productInCart: [
        {
            id: "",
            title: "",
            image: "",
            price: 0,
            rating: 0,
        },
    ],
    currentTotalAmount: 0,
    currentProductNum: 0,
};

export const reducers = {
    updateProductInCartAction: (
        _state: UITop,
        action: PayloadAction<CartProduct>
    ) => ({
        ..._state,
        productInCart: [..._state.productInCart, action.payload],
    }),
    updateCurrentTotalAmountAction: (
        _state: UITop,
        action: PayloadAction<number>
    ) => ({
        ..._state,
        currentTotalAmount: _state.currentTotalAmount + action.payload,
    }),
    updateCurrentProductNumAction: (_state: UITop) => ({
        ..._state,
        currentProductNum: _state.currentProductNum + 1,
    }),
};
