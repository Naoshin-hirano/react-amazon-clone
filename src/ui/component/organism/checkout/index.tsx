import React from "react";
import "../../../../design/component/atoms/checkout/index.css";
import { Subtotal } from "./subtotal";
import { CheckoutProduct } from "../common/checkout-product";
import { CartProduct } from "core/store/ui/top/type";

export const Checkout: React.FC<any> = (props) => {
    const { busketInfo } = props;
    return (
        <div className="checkout">
            <div className="checkout__left">
                <img
                    className="checkout__ad"
                    src="https://images-na.ssl-images-amazon.com/images/G/02/UK_CCMP/TM/OCC_Amazon1._CB423492668_.jpg"
                    alt="checkout component ad banner"
                />

                <div>
                    <h3>平野直弥さん</h3>
                    <h2 className="checkout__title">カート内の商品</h2>

                    {busketInfo.productInCart ? (
                        busketInfo.productInCart.map(
                            (item: CartProduct, key: string) => (
                                <CheckoutProduct
                                    key={key}
                                    id={item.id}
                                    title={item.title}
                                    image={item.image}
                                    price={item.price}
                                    rating={item.rating}
                                />
                            )
                        )
                    ) : (
                        <>商品が存在しません</>
                    )}
                </div>
            </div>

            <div className="checkout__right">
                <Subtotal busketInfo={busketInfo} />
            </div>
        </div>
    );
};
