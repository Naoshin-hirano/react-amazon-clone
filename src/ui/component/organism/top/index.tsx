import React from "react";
import "../../../../design/component/atoms/top/index.css";
import { Product } from "./product";
import Carousel from "react-material-ui-carousel";
import Paper from "@material-ui/core/Paper";
import { images } from "./images";

export const Top: React.FC<any> = () => {
    return (
        <div className="home">
            <div className="home__container">
                <div className="home__image">
                    <Carousel
                        navButtonsAlwaysVisible={false}
                        indicators={false}
                        fullHeightHover={false}
                    >
                        {images.map((image, i) => (
                            <Image key={i} image={image} />
                        ))}
                    </Carousel>
                </div>

                <div className="home__rowContainer">
                    <div className="home__row">
                        <Product
                            id="12321341"
                            title="The Lean Startup: How Constant Innovation Creates Radically Successful Businesses Paperback"
                            price={1196}
                            rating={5}
                            image="https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._SX325_BO1,204,203,200_.jpg"
                        />
                        <Product
                            id="49538094"
                            title="Kenwood kMix Stand Mixer for Baking, Stylish Kitchen Mixer with K-beater, Dough Hook and Whisk, 5 Litre Glass Bowl"
                            price={239}
                            rating={4}
                            image="https://images-na.ssl-images-amazon.com/images/I/81O%2BGNdkzKL._AC_SX450_.jpg"
                        />
                    </div>

                    <div className="home__row">
                        <Product
                            id="4903850"
                            title="Samsung LC49RG90SSUXEN 49' Curved LED Gaming Monitor"
                            price={199}
                            rating={3}
                            image="https://images-na.ssl-images-amazon.com/images/I/71Swqqe7XAL._AC_SX466_.jpg"
                        />
                        <Product
                            id="23445930"
                            title="Amazon Echo (3rd generation) | Smart speaker with Alexa, Charcoal Fabric"
                            price={98}
                            rating={5}
                            image="https://media.very.co.uk/i/very/P6LTG_SQ1_0000000071_CHARCOAL_SLf?$300x400_retinamobilex2$"
                        />
                        <Product
                            id="3254354345"
                            title="New Apple iPad Pro (12.9-inch, Wi-Fi, 128GB) - Silver (4th Generation)"
                            price={598}
                            rating={4}
                            image="https://images-na.ssl-images-amazon.com/images/I/816ctt5WV5L._AC_SX385_.jpg"
                        />
                    </div>

                    <div className="home__row">
                        <Product
                            id="90829332"
                            title="Samsung LC49RG90SSUXEN 49' Curved LED Gaming Monitor - Super Ultra Wide Dual WQHD 5120 x 1440"
                            price={1094}
                            rating={4}
                            image="https://images-na.ssl-images-amazon.com/images/I/6125mFrzr6L._AC_SX355_.jpg"
                        />
                    </div>
                </div>
            </div>
        </div>
    );
};

const Image: React.FC<any> = ({ image }) => {
    return (
        <Paper>
            <img src={image.src} alt="amazon banner" />
        </Paper>
    );
};
