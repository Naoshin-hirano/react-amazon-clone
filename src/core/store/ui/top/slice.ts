import { createSlice } from "@reduxjs/toolkit";
import { initialState, reducers } from "./service";

export const UITopSlice = createSlice({
    name: "ui/top",
    initialState,
    reducers,
});
