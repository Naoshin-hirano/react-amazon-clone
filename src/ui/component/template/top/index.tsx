import React from "react";
import { Header } from "ui/component/organism/common/header";
import { Top } from "ui/component/organism/top";

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Header />
            <Top {...props} />
        </div>
    );
};
