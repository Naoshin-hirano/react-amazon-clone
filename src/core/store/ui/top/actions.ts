import { UITopSlice } from "./slice";

export const UITopAction = UITopSlice.actions;
