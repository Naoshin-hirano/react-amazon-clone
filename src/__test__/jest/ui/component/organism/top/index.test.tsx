import { render } from "@testing-library/react";
import React from "react";
import { Top } from "ui/component/organism/top";
// import * as Store from "core/store";

const mockHistoryPush = jest.fn();
// const dispatchSpy = jest.spyOn(Store, "dispatch");

jest.mock("@tanstack/react-location", () => {
    return {
        ...jest.requireActual("@tanstack/react-location"),
        useLocation: () => ({
            history: {
                push: mockHistoryPush,
            },
        }),
    };
});

describe("Header Unit Test", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    type TopTitle = {
        title: string;
    };

    const topTitle: TopTitle = {
        title: "トップタイトル",
    };

    it("カートへボタン押下", async () => {
        const { asFragment } = render(<Top topTitle={topTitle} />);
        expect(asFragment()).toMatchSnapshot("Top");
    });

    // it("カートへボタン押下", async () => {
    //     const wrapper = render(<Top topTitle={topTitle} />);
    //     await act(async () => {
    //         const buttonName = await wrapper.getByText(/カートへ/);
    //         await fireEvent.click(buttonName);
    //     });
    //     expect(dispatchSpy).toHaveBeenCalledTimes(0);
    //     expect(mockHistoryPush).toHaveBeenCalledTimes(0);
    // });
});
