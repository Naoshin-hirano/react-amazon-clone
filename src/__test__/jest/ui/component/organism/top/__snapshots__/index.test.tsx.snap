// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Unit Test カートへボタン押下: Top 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="home__container"
    >
      <div
        class="home__image"
      >
        <div
          class="Carousel-root-1 "
        >
          <div
            class="CarouselItem"
          >
            <div
              style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <img
                  alt="amazon banner"
                  src="https://images-eu.ssl-images-amazon.com/images/G/02/digital/video/merch2016/Hero/Covid19/Generic/GWBleedingHero_ENG_COVIDUPDATE__XSite_1500x600_PV_en-GB._CB428684220_.jpg"
                />
              </div>
            </div>
          </div>
          <div
            class="Carousel-buttonWrapper-6   Carousel-next-12"
          >
            <button
              aria-label="Next"
              class="MuiButtonBase-root MuiIconButton-root Carousel-button-11 Carousel-buttonHidden-10  "
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="Carousel-buttonWrapper-6   Carousel-prev-13"
          >
            <button
              aria-label="Previous"
              class="MuiButtonBase-root MuiIconButton-root Carousel-button-11 Carousel-buttonHidden-10  "
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="home__rowContainer"
      >
        <div
          class="home__row"
        >
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                The Lean Startup: How Constant Innovation Creates Radically Successful Businesses Paperback
              </p>
              <p>
                12321341
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  1196
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._SX325_BO1,204,203,200_.jpg"
            />
            <button>
              カートへ
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                Kenwood kMix Stand Mixer for Baking, Stylish Kitchen Mixer with K-beater, Dough Hook and Whisk, 5 Litre Glass Bowl
              </p>
              <p>
                49538094
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  239
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://images-na.ssl-images-amazon.com/images/I/81O%2BGNdkzKL._AC_SX450_.jpg"
            />
            <button>
              カートへ
            </button>
          </div>
        </div>
        <div
          class="home__row"
        >
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                Samsung LC49RG90SSUXEN 49' Curved LED Gaming Monitor
              </p>
              <p>
                4903850
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  199
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://images-na.ssl-images-amazon.com/images/I/71Swqqe7XAL._AC_SX466_.jpg"
            />
            <button>
              カートへ
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                Amazon Echo (3rd generation) | Smart speaker with Alexa, Charcoal Fabric
              </p>
              <p>
                23445930
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  98
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://media.very.co.uk/i/very/P6LTG_SQ1_0000000071_CHARCOAL_SLf?$300x400_retinamobilex2$"
            />
            <button>
              カートへ
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                New Apple iPad Pro (12.9-inch, Wi-Fi, 128GB) - Silver (4th Generation)
              </p>
              <p>
                3254354345
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  598
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://images-na.ssl-images-amazon.com/images/I/816ctt5WV5L._AC_SX385_.jpg"
            />
            <button>
              カートへ
            </button>
          </div>
        </div>
        <div
          class="home__row"
        >
          <div
            class="product"
          >
            <div
              class="product__info"
            >
              <p>
                Samsung LC49RG90SSUXEN 49' Curved LED Gaming Monitor - Super Ultra Wide Dual WQHD 5120 x 1440
              </p>
              <p>
                90829332
              </p>
              <p
                class="product__price"
              >
                <small>
                  ¥
                </small>
                <strong>
                  1094
                </strong>
              </p>
              <div
                class="product__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
            </div>
            <div
              class="product__rating"
            />
            <img
              alt="The Lean Startup book cover"
              src="https://images-na.ssl-images-amazon.com/images/I/6125mFrzr6L._AC_SX355_.jpg"
            />
            <button>
              カートへ
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
